lib/rbs_rails/active_record.rb

```diff
@@ -366,20 +366,44 @@ def authenticate_#{attribute}: (String) -> (#{klass_name} | false)
           next unless node.children[0].nil?
           next unless node.children[1] == :enum
 
-          definitions = node.children[2]
-          next unless definitions
-          next unless definitions.type == :hash
-          next unless traverse(definitions).all? { |n| [:str, :sym, :int, :hash, :pair, :true, :false].include?(n.type) }
+          if Rails::VERSION::MAJOR >= 7
```

Comment:
Personally, I'd like to separate the Enum definition parser to the independent class (ex. `RbsRails::ActiveRecord::Enum`). But I kept this as is to make a pull request smaller to review easily. I'll post another PR for the refactoring if we'll agree.